dials.go

```diff
@@ -144,18 +143,50 @@ type Decoder interface {
 	Decode(io.Reader, *Type) (reflect.Value, error)
 }
 
-type watchTab struct {
+type valueUpdate struct {
 	source Source
 	value  reflect.Value
 }
 
+func (valueUpdate) isStatusReport() {}
+
+type watchStatusUpdate interface {
+	isStatusReport()
+}
+
+type watchArgs struct {
+	s Source
+	c chan watchStatusUpdate
+}
+
+// NewValue reports a new value. Returns an error if the internal
+// reporting channel is full and the context expires/is-canceled.
+func (w *watchArgs) NewValue(ctx context.Context, val reflect.Value) error {
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	case w.c <- &valueUpdate{source: w.s, value: val}:
+		return nil
+	}
+}
+
+// WatchArgs provides methods for a Watcher implementation to update the state
+// of a Dials instance.
+type WatchArgs interface {
+	// NewValue reports a new value. The base implementation returns an
+	// error if the internal reporting channel is full and the context
+	// expires/is-canceled, however, wrapping implementations are free to
+	// return any other error as appropriate.
+	NewValue(ctx context.Context, val reflect.Value) error
```

Comment:
I can appreciate the desire for brevity, but every time I see this method name I think it's creating a new value.  Maybe it's worth the few extra characters and call is `ReportNewValue` or alternatively `Notify`?

Reply:
Thanks!